src/nedap/speced/def/impl/type_hinting.cljc

```diff
@@ -91,17 +93,17 @@
 
 (defn type-hint?
   ([x]
-   (assert (not= *ns* this-ns) "For an accurate `resolve` call (see `#'clj-type-hint?`).")
    #?(:clj  (clj-type-hint? x)
       :cljs (cljs-type-hint? x)))
 
   ([x clj?]
-   (assert (not= *ns* this-ns) "For an accurate `resolve` call (see `#'clj-type-hint?`).")
    (assert (boolean? clj?))
    #?(:clj  (if clj?
               (clj-type-hint? x)
               (cljs-type-hint? x))
-      :cljs (assert false))))
+      :cljs (do
+              (assert (not clj?))
```

Comment:
are you testing the clj-reader here?

Reply:
Not necessarily: at some point we might add implement support for self-hosted cljs.

Things could break in strange ways (reader conditionals in macros behave differently in self-hosted cljs), hence the heavy asserting.